ring_buffer/ring_buffer.py

```diff
@@ -8,25 +8,89 @@ def __init__(self, capacity):
         self.storage = DoublyLinkedList()
 
     def append(self, item):
-        pass
+        if not self.current:
+            # If self.current is None, the list isn't full yet
+            self.storage.add_to_tail(item)
+            if self.storage.length == self.capacity:
+                # If this puts the list at capacity, set self.current
```

Comment:
I definitely think there's a way to make this a little cleaner. I'm happy to share the solution from the Lambda repo or the one I have but I think if you sleep it over and come back to this you might find you were over complicating it more than it needed to be.

Reply:
Remember... we don't have to replace the ENTIRE node and change/make new pointers. We could just update the value of the REFERENCE to the object in self.current, which is basically a pointer for our LRU (least recently used) item.